rpaas/consul_manager.py

```diff
@@ -198,6 +193,39 @@ def list_upstream(self, instance_name, upstream_name):
     def _save_upstream(self, instance_name, upstream_name, servers):
         self.client.kv.put(self._upstream_key(instance_name, upstream_name), ",".join(servers))
 
+    def find_acl_network(self, instance_name, src=None):
+        src = self._normalize_acl_src(src)
+        acls = self.client.kv.get(self._acl_key(instance_name, src), recurse=True)[1]
+        if not acls:
+            return []
+        acls_list = []
+        for acl in acls:
+            src = acl['Key'].split('/')[-1]
+            acls_list.append({"source": self._normalize_acl_src(src),
+                              "destination": acl["Value"].split(",")})
+        return acls_list
+
+    def store_acl_network(self, instance_name, src, dst):
+        acls = self.find_acl_network(instance_name, src)
+        if acls:
+            acls = set(acls[0]['destination'])
+            acls |= set([dst])
+        else:
+            acls.append(dst)
+        src = self._normalize_acl_src(src)
+        self.client.kv.put(self._acl_key(instance_name, src), ",".join(acls))
```

Comment:
Since we're not using atomic operations on redis anymore maybe we should set `cas=<modifyIndex from find_acl_network request>` when updating the key in consul. (And maybe retry on failure? Not sure about that) This would prevent inconsistencies on concurrent requests.

Reply:
I think using rpaas Lock class, which uses redis global lock, is a better approach than using consul last modified index. Even remotely unlikely to happen race in this case, who knows :)